src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1512,7 +1512,7 @@ internal static bool TryCreate(int year, int month, int day, int hour, int minut
         //
 
         // /// <inheritdoc cref="IAdditionOperators{TSelf, TOther, TResult}.op_Addition(TSelf, TOther)" />
-        // static DateTime IAdditionOperators<DateTime, TimeSpan, DateTime>.operator checked +(DateTime left, TimeSpan right) => checked(left + right);
+        static DateTime IAdditionOperators<DateTime, TimeSpan, DateTime>.operator checked +(DateTime left, TimeSpan right) => left + right;
```

Comment:
This is interesting.  DateTime.MaxValue + TimeSpan.FromMilliseconds(1) will throw an exception, regardless of checked.  In a generic math context, that feels a little strange, but I'm not sure there's a better answer (as long as we continue to want these only-slightly-math-related types implementing these interfaces).

Reply:
I think this is an area where we'll want to consider the implications of there now being `user-defined checked operators` for certain core types (like `DateTime` and similar).

One could imagine us "properly" supporting checked/unchecked contexts for these types now. One could also imagine us continuing to always throw on overflow.

`checked(expr)` doesn't necessarily mean that `expr` will throw on overflow; and `unchecked(expr)` doesn't necessarily mean that `expr` will not throw on overflow. It really depends on the type and what operations it decides to support.

We'll probably want to decide on guidance as well, including the implications of taking the behavioral break for already compiled dependents.

Reply:
We also have the option of saying:

`DateTime` always overflows when used as a concrete type but when used in a generic math context it properly respects `checked`/`unchecked`. We can always have the public APIs and the explicit interface implementations SxS, just needing to have API review in agreement that's overall good/desirable

Reply:
Right.  Seems like we basically have four options:
1. Don't implement the generic math interfaces on these types.
2. Leave the behavior as is.  Someone using these types in a generic context might experience exceptions for "overflow" (but not actually OverflowException) when not in unchecked.
3. Implement the new unchecked operators on these types to not throw.  This means someone using these in a generic context will have more consistent behavior regardless of the concrete type being used, but it yields an inconsistency between how the type behaves when using it directly vs generically.
4. (3) and change the existing behavior of the existing operators to be non-throwing.  That's a breaking change in that someone could easily be relying on that behavior to stop "bad things" from happening.

We don't need to decide now, but please ensure there's an issue open to track making a decision around this for .NET 7.

Reply:
Will log an issue tracking this before merging.

Reply:
https://github.com/dotnet/runtime/issues/67744